rest_framework_json_api/backends.py

```diff
@@ -0,0 +1,148 @@
+from django.db.models.sql.constants import ORDER_PATTERN
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework.exceptions import ValidationError
+from rest_framework.filters import (BaseFilterBackend, OrderingFilter,
+                                    SearchFilter)
+from rest_framework.settings import api_settings
+
+
+class JsonApiFilterMixin(object):
+    """
+    class to share data among filtering backends
+    """
+    jsonapi_query_keywords = ('sort', 'filter', 'fields', 'page', 'include')
+    search_param = api_settings.SEARCH_PARAM
+    ordering_param = 'sort'
+
+    def __init__(self):
+        self.filter_keys = []
+
+
+class JsonApiQueryValidationFilter(JsonApiFilterMixin, BaseFilterBackend):
+    """
+    A backend filter that validates query parameters for jsonapi spec conformance and raises a 400 error
+    rather than silently ignoring unknown parameters or incorrect usage.
+
+    set `allow_duplicate_filters = True` if you are OK with the same filter being repeated.
+
+    TODO: For jsonapi error object compliance, must set jsonapi errors "parameter" for the ValidationError.
+          This requires extending DRF/DJA Exceptions.
+    """
+    allow_duplicated_filters = False
+
+    def validate_query_params(self, request):
+        """
+        Validate that query params are in the list of valid `jsonapi_query_keywords`
+        Raises ValidationError if not.
+        """
+        for qp in request.query_params.keys():
+            bracket = qp.find('[')
```

Comment:
I was trying to care about performance as this code executes for every request. Also the match would be something more complicated since it's searching for all keywords, whether or not they have brackets (e.g. `filter[foo], sort, ...`). It seems a regex is overkill and detracts from readability.